scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -846,17 +876,27 @@ object Execution {
     FlowDefExecution(fn)
 
   def forceToDisk[T](t: TypedPipe[T]): Execution[TypedPipe[T]] =
+    fromFuture(RandomUUID()).flatMap(ForceToDisk(t))
+
+  def toIterable[T](t: TypedPipe[T]): Execution[Iterable[T]] =
+    fromFuture(RandomUUID()).flatMap(ToIterable(t))
+
+  def toIterableWithClose[T](t: TypedPipe[T]): Execution[(Iterable[T], Execution[Unit])] =
+    fromFuture(RandomUUID()).flatMap(ToIterableWithClose(t))
```

Comment:
Actually, I think we need something like: `Execution.withNewCache(fromFuture(RandomUUID())).flatMap...`

Because I think it is essential that we not cache the UUID generation during retries.

I think this is the answer to my question about the possible change in semantics of having the UUID.random moved out of the Execution evaluation earlier.